tracks/bash/exercises/proverb/mentoring.md

```diff
@@ -0,0 +1,44 @@
+### Reasonable Solutions
+```bash
+#!/usr/bin/env bash
+
+main() {
+  if [ $# -eq 0 ]; then
+    echo ""
+    exit
+  fi
+  for ((currentNounIndex = 1, nextNounIndex = 2; currentNounIndex < $#; currentNounIndex++, nextNounIndex++)); do
+    proverb+=$"For want of a ${!currentNounIndex} the "${!nextNounIndex}" was lost.\n"
+  done
+  proverb+="And all for the want of a $1."
+  echo -e "$proverb"
+}
+
+main "$@"
+```
+
+### Common suggestions
+- use "$#" to get number of arguments
+This will even handle the whitespace tests, such as:
+
+```
+bash proverb.sh "rusty nail" "horse shoe"
+
+output:
+
+$# = 2
+$1 = "rusty nail"
+$2 = "horse shoe"
+```
```

Comment:
```suggestion
\```shell
bash proverb.sh "rusty nail" "horse shoe"
\```
output:

\```
$# = 2
$1 = "rusty nail"
$2 = "horse shoe"
\```
```

Sorry, not sure how to make that cleaner, since the backticks are part of it, and it is being rendered otherwise.